tests/codegen/issues/issue-93036-assert-index.rs

```diff
@@ -0,0 +1,17 @@
+//@ compile-flags: -O
+//@ min-llvm-version: 17
+
+#![crate_type = "lib"]
+
+#[no_mangle]
+// CHECK-LABEL: @foo
+// CHECK: {{.*}}:
+// CHECK: ret
```

Comment:
If you have CHECK-NOT: unreachable, I think you want to remove this CHECK: ret? Otherwise the case where there is unreachable followed by ret (rather than ret followed by unreachable) will still be accepted.